test/common/wpt.js

```diff
@@ -527,47 +593,28 @@ class WPTRunner {
   // TODO(joyeecheung): work with the upstream to port more tests in .html
   // to .js.
   async runJsTests() {
-    let queue = [];
-
-    // If the tests are run as `node test/wpt/test-something.js subset.any.js`,
-    // only `subset.any.js` will be run by the runner.
-    if (process.argv[2]) {
-      const filename = process.argv[2];
-      if (!this.specMap.has(filename)) {
-        throw new Error(`${filename} not found!`);
-      }
-      queue.push(this.specMap.get(filename));
-    } else {
-      queue = this.buildQueue();
-    }
+    const queue = this.buildQueue();
 
-    this.inProgress = new Set(queue.map((spec) => spec.filename));
+    const run = limit(os.availableParallelism());
 
     for (const spec of queue) {
-      const testFileName = spec.filename;
       const content = spec.getContent();
-      const meta = spec.meta = this.getMeta(content);
+      const meta = spec.getMeta(content);
 
       const absolutePath = spec.getAbsolutePath();
       const relativePath = spec.getRelativePath();
       const harnessPath = fixtures.path('wpt', 'resources', 'testharness.js');
       const scriptsToRun = [];
-      let needsGc = false;
 
       // Scripts specified with the `// META: script=` header
-      if (meta.script) {
-        for (const script of meta.script) {
-          if (script === '/common/gc.js') {
-            needsGc = true;
-          }
-          const obj = {
-            filename: this.resource.toRealFilePath(relativePath, script),
-            code: this.resource.read(relativePath, script, false),
-          };
-          this.scriptsModifier?.(obj);
-          scriptsToRun.push(obj);
-        }
-      }
+      meta.script?.forEach((script) => {
+        const obj = {
+          filename: this.resource.toRealFilePath(relativePath, script),
+          code: this.resource.read(relativePath, script),
+        };
+        this.scriptsModifier?.(obj);
+        scriptsToRun.push(obj);
+      });
```

Comment:
optional nit: I would write it with a `.map`, to make it clear how the array is filled up
```suggestion
      const scriptsToRun = meta.script?.map((script) => {
        const obj = {
          filename: this.resource.toRealFilePath(relativePath, script),
          code: this.resource.read(relativePath, script),
        };
        this.scriptsModifier?.(obj);
        return obj;
      }) ?? [];
```